internal/gengapic/client_init.go

```diff
@@ -139,193 +139,405 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
+	p("func New%[1]sClient(ctx context.Context, opts ...option.ClientOption) (*%[1]sClient, error) {", servName)
+	p("  clientOpts := default%[1]sGrpcCallOptions()", servName)
+
+	p("  if new%sGrpcClientHook != nil {", servName)
+	p("    hookOpts, err := new%sGrpcClientHook(ctx, clientHookParams{})", servName)
+	p("    if err != nil {")
+	p("      return nil, err")
+	p("    }")
+	p("    clientOpts = append(clientOpts, hookOpts...)")
+	p("  }")
+	p("")
+	p("  disableDeadlines, err := checkDisableDeadlines()")
+	p("  if err != nil {")
+	p("    return nil, err")
+	p("  }")
+	p("")
+	p("  connPool, err := gtransport.DialPool(ctx, append(clientOpts, opts...)...)")
+	p("  if err != nil {")
+	p("    return nil, err")
+	p("  }")
+	p("  c := &%sGrpcClient{", lowcaseServName)
+	p("    connPool:    connPool,")
+	p("    disableDeadlines: disableDeadlines,")
+	p("    %s: %s.New%sClient(connPool),", grpcClientField(servName), imp.Name, serv.GetName())
+	p("  }")
+	p("  c.setGoogleClientInfo()")
+	p("")
+
+	if hasRPCForLRO {
+		p("  c.LROClient, err = lroauto.NewOperationsClient(ctx, gtransport.WithConnPool(connPool))")
 		p("  if err != nil {")
+		p("    // This error \"should not happen\", since we are just reusing old connection pool")
+		p("    // and never actually need to dial.")
+		p("    // If this does happen, we could leak connp. However, we cannot close conn:")
+		p("    // If the user invoked the constructor with option.WithGRPCConn,")
+		p("    // we would close a connection that's still in use.")
+		p("    // TODO: investigate error conditions.")
 		p("    return nil, err")
 		p("  }")
-		p("  c := &%sClient{", servName)
-		p("    connPool:    connPool,")
-		p("    disableDeadlines: disableDeadlines,")
-		p("    CallOptions: default%sCallOptions(),", servName)
+	}
+
+	if g.hasLROMixin() {
+		p("  c.operationsClient = longrunningpb.NewOperationsClient(connPool)")
 		p("")
-		p("    %s: %s.New%sClient(connPool),", grpcClientField(servName), imp.Name, serv.GetName())
-		p("  }")
-		p("  c.setGoogleClientInfo()")
+	}
+
+	if g.hasIAMPolicyMixin() {
+		p("  c.iamPolicyClient = iampb.NewIAMPolicyClient(connPool)")
 		p("")
+	}
 
-		if hasRPCForLRO {
-			p("  c.LROClient, err = lroauto.NewOperationsClient(ctx, gtransport.WithConnPool(connPool))")
-			p("  if err != nil {")
-			p("    // This error \"should not happen\", since we are just reusing old connection pool")
-			p("    // and never actually need to dial.")
-			p("    // If this does happen, we could leak connp. However, we cannot close conn:")
-			p("    // If the user invoked the constructor with option.WithGRPCConn,")
-			p("    // we would close a connection that's still in use.")
-			p("    // TODO: investigate error conditions.")
-			p("    return nil, err")
-			p("  }")
-		}
+	if g.hasLocationMixin() {
+		p("  c.locationsClient = locationpb.NewLocationsClient(connPool)")
+		p("")
+	}
+	p("  return &%sClient{internal%[1]sClient: c, CallOptions: clientOpts}, nil", servName)
```

Comment:
`CallOptions` should be supplied the `defaultCallOptions` not the `ClientOption`s. They are two different things, as mentioned above.